src/Common/tests/System/Collections/IList.NonGeneric.Tests.cs

```diff
@@ -1049,5 +1050,40 @@ public void IList_NonGeneric_RemoveAt_ZeroMultipleTimes(int count)
         }
 
         #endregion
+
+        #region Enumerator.Current
+
+        // Enumerator.Current should fail at end after new elements was added
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void IList_NonGeneric_CurrentAtEnd_AfterAdd(int count)
+        {
+            if (!IsReadOnly && !ExpectedFixedSize && Enumerator_Current_UndefinedOperation_Throws)
+            {
+                IList collection = NonGenericIListFactory(count);
+                IEnumerator enumerator = collection.GetEnumerator();
+                while (enumerator.MoveNext()) ; // Go to end of enumerator
+                Assert.Throws<InvalidOperationException>(() => enumerator.Current); // Enumerator.Current should fail
+
+                int seed = 523561;
+                collection.Add(CreateT(seed++));
+
+                // (List<T>.Enumerator as IEnumerator).Current have breaking issue here
+                bool isGenericList = false;
+                {
+                    var typeInfo = enumerator.GetType().GetTypeInfo();
+                    isGenericList = typeInfo.IsGenericType && typeInfo.GetGenericTypeDefinition() == typeof(List<>.Enumerator);
+                }
```

Comment:
Same comment here... we shouldn't be special-casing specific types in these tests.